capellambse/_native.py

```diff
@@ -144,28 +148,86 @@ def __call__(
         project: pathlib.Path,
         /,
         *args: str | pathlib.Path,
-    ) -> subprocess.CompletedProcess:
-        cmd: list[str | pathlib.Path] = [
-            "docker",
-            "run",
-            "--rm",
-            "--detach=false",
-            f"-v{workspace}:/workspace",
-            f"-v{project}:/model",
-            self.image,
-            *_COMMON_ARGS,
-            _ARG_WORKSPACE,
-            "/workspace",
-            _ARG_PROJECT,
-            "/model/main_model",
-            *args,
-        ]
-        _LOGGER.debug("Running Capella in Docker with command %r", cmd)
-        return subprocess.run(
-            cmd,
-            stdin=subprocess.DEVNULL,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
+    ):
+        entrypoint = self.get_entrypoint()
+        container_name = "".join(
+            random.choice(string.ascii_lowercase) for i in range(5)
+        )
+        with subprocess.Popen(
```

Comment:
Style improvement: It's a bit inconsistent to have some commands be direct `subprocess.Popen / run` calls here and others be methods which wrap them. With the length of a subprocess call, IMO it makes most sense to extract all of them out, which should make this `__call__` method much more pleasant to read.